services/rsources/handler.go

```diff
@@ -111,22 +111,16 @@ func (sh *sourcesHandler) AddFailedRecords(ctx context.Context, tx *sql.Tx, jobR
 	if sh.config.SkipFailedRecordsCollection {
 		return nil
 	}
-
-	row := tx.QueryRow(`WITH new_key AS (
-		INSERT INTO rsources_failed_keys_v2 (id, job_run_id, task_run_id, source_id, destination_id)
+	row := tx.QueryRow(`INSERT INTO rsources_failed_keys_v2 (id, job_run_id, task_run_id, source_id, destination_id)
 		VALUES ($1, $2, $3, $4, $5) 
-		ON CONFLICT (job_run_id, task_run_id, source_id, destination_id, db_name) DO NOTHING
-		RETURNING id
-	) SELECT COALESCE(
-		(SELECT id FROM new_key),
-		(SELECT id FROM rsources_failed_keys_v2 WHERE job_run_id = $2 AND task_run_id = $3 AND source_id = $4 AND destination_id = $5)
```

Comment:
> (SELECT id FROM rsources_failed_keys_v2 WHERE job_run_id = $2 AND task_run_id = $3 AND source_id = $4 AND destination_id = $5)

Curious, about why this got removed?

Reply:
it is not needed anymore, since this will always return the appropriate id
```sql
INSERT INTO rsources_failed_keys_v2 
  (id, job_run_id, task_run_id, source_id, destination_id)
  VALUES 
  ($1, $2, $3, $4, $5) 
  ON CONFLICT (job_run_id, task_run_id, source_id, destination_id, db_name) 
  DO UPDATE SET ts = NOW()
  RETURNING id
```